components/event-sources/reconciler/object/service.go

```diff
@@ -77,6 +77,23 @@ func WithMinScale(replicas int) ObjectOption {
 	}
 }
 
+// WithSpecTemplateLabel sets the value of an API object's metadata label
+func WithSpecTemplateLabel(key, val string) ObjectOption {
+	return func(o metav1.Object) {
+		s := o.(*servingv1alpha1.Service)
+
+		tpl := &s.Spec.ConfigurationSpec.Template
+		if *tpl == nil {
+			*tpl = &servingv1alpha1.RevisionTemplateSpec{}
+		}
+		obj := (*tpl).ObjectMeta
+		if obj.Labels == nil {
+			obj.Labels = make(map[string]string)
+		}
+		obj.Labels[key] = val
```

Comment:
Won't work because `obj` is a _copy_ of ObjectMeta, not the original ObjectMeta. https://play.golang.org/p/D4vazyZiZcU
You have to work with pointers, or just use `(*tpl).ObjectMeta` directly.